downstairs/src/lib.rs

```diff
@@ -3350,6 +3351,72 @@ pub async fn start_downstairs(
     Ok(join_handle)
 }
 
+/// Clone the extent files in a region from another running downstairs.
+///
+/// Use the reconcile/repair extent methods to copy another downstairs.
+/// The source downstairs must have the same RegionDefinition as we do,
+/// and both downstairs must be running in read only mode.
+pub async fn clone_region(
+    d: Arc<Mutex<Downstairs>>,
+    source: SocketAddr,
+) -> Result<()> {
+    let info = crucible_common::BuildInfo::default();
+    let log = d.lock().await.log.new(o!("task" => "clone".to_string()));
+    info!(log, "Crucible Version: {}", info);
+    info!(
+        log,
+        "Upstairs <-> Downstairs Message Version: {}", CRUCIBLE_MESSAGE_VERSION
+    );
+
+    info!(log, "Connecting to {source} to obtain our extent files.");
+
+    let url = format!("http://{:?}", source);
+    let repair_server = Client::new(&url);
+    let source_def = match repair_server.get_region_info().await {
+        Ok(def) => def.into_inner(),
+        Err(e) => {
+            bail!("Failed to get source region definition: {e}");
+        }
+    };
+    info!(log, "The source RegionDefinition is: {:?}", source_def);
+
+    let source_ro_mode = match repair_server.get_region_mode().await {
+        Ok(ro) => ro.into_inner(),
+        Err(e) => {
+            bail!("Failed to get source mode: {e}");
+        }
+    };
+
+    info!(log, "The source mode is: {:?}", source_ro_mode);
+    if !source_ro_mode {
+        bail!("Source downstairs is not read only");
+    }
+
+    let mut ds = d.lock().await;
+
+    let my_def = ds.region.def();
+    info!(log, "my def is {:?}", my_def);
+
+    if let Err(e) = my_def.compatible(source_def) {
+        bail!("Incompatible region definitions: {e}");
+    }
+
+    if let Err(e) = ds.region.close_all_extents().await {
+        bail!("Failed to close all extents: {e}");
+    }
```

Comment:
At this point, do we need to reopen them? If we try to run the clone again, will it fail?

Reply:
The clone region subcommand either runs to completion, or exits with an error.
So if we fail here, we are exiting the program.  If we run  the `clone` again, it will
start from extent 0 replace everything, even if a previous run only made it partially
through the list.

Reply:
I was worried that reissuing `clone` would hit

https://github.com/oxidecomputer/crucible/blob/2d7e499a9dfe5a6c287f70d63beb676c30bfbb3d/downstairs/src/region.rs#L467

but if failing will exit the program, then rerunning it again will start fresh. Makes sense :)